### PR TITLE
Add CreateFlexibleBlueprint method to dp-api-clients-go

### PR DESCRIPTION
**Pull request by goofballLogic**
### What

Added CreateFlexibleBlueprint function to accept population type field
Refactored code to share implementation between CreateFlexibleBlueprint and CreateBlueprint
Corrected tests for CreateBlueprint
Added tests for CreateFlexibleBlueprint

### How to review

Refactoring is sound
New functionality is present
Tests cover functionality

### Who can review

gophers other than @goofballLogic

### CODE REVIEW

File: filter/data.go [hunk 1]
```diff
@@ -39,13 +39,23 @@ type DimensionOptions struct {
 	TotalCount int               `json:"total_count"`
 }
 
-// CreateBlueprint represents the fields required to create a filter blueprint
-type CreateBlueprint struct {
+// createBlueprint represents the fields required to create a filter blueprint
```
Comment (goofballLogic) [inline filter/data.go:42]: this was found to be exported in error so changed

File: filter/filter.go [hunk 2]
```diff
@@ -422,27 +422,78 @@ func (c *Client) GetDimensionOptionsBatchProcess(ctx context.Context, userAuthTo
...
 	}
 
+	if err = json.Unmarshal(respBody, &cb); err != nil {
+		return "", "", err
+	}
+
+	return cb.FilterID, eTag, nil
+}
+
+func (c *Client) postBlueprint(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceToken, collectionID, datasetID, edition, version string, reqBody []byte) ([]byte, string, error) {
```
Comment (goofballLogic) [inline filter/filter.go:495]: this function is behaviour required for POSTing to the `/filters` endpoint common to both the above methods

File: filter/filter_test.go [hunk 3]
```diff
@@ -815,19 +816,119 @@ func TestClient_GetDimensionOptionsInBatches(t *testing.T) {
...
+
 	checkRequest := func(httpClient *dphttp.ClienterMock, expectedFilterID string) {
 		So(len(httpClient.DoCalls()), ShouldEqual, 1)
 
 		actualBody, _ := ioutil.ReadAll(httpClient.DoCalls()[0].Req.Body)
-		var actualVersion CreateBlueprint
+		var actualVersion createBlueprint
 		err := json.Unmarshal(actualBody, &actualVersion)
 		So(err, ShouldBeNil)
+		So(actualVersion, ShouldResemble, expectedRequest)
```
Comment (goofballLogic) [inline filter/filter_test.go:931]: previously tests were not checking that the request body was correct

**Review by jayfreestone — APPROVED** [review]
👍 LGTM

**Pull request merged**